x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/action/profile/ProfileHasPrivilegesRequestTests.java

```diff
@@ -0,0 +1,212 @@
+        assertThat(exception, notNullValue());
+        assertThat(exception.validationErrors(), hasItem("Application names may not contain '*' (found '*')"));
+    }
+
+    public static PrivilegesToCheck randomValidPrivilegesToCheckRequest() {
+        String[] clusterPrivileges = randomClusterPrivileges();
+        RoleDescriptor.IndicesPrivileges[] indicesPrivileges = randomIndicesPrivileges();
+        RoleDescriptor.ApplicationResourcePrivileges[] appPrivileges = randomApplicationResourcePrivileges();
+        return new PrivilegesToCheck(clusterPrivileges, indicesPrivileges, appPrivileges);
+    }
+
+    private static String[] randomClusterPrivileges() {
+        return randomSubsetOf(randomIntBetween(1, 5), ClusterPrivilegeResolver.names()).toArray(new String[0]);
+    }
+
+    private static RoleDescriptor.IndicesPrivileges[] randomIndicesPrivileges() {
+        RoleDescriptor.IndicesPrivileges[] indicesPrivileges = new RoleDescriptor.IndicesPrivileges[randomIntBetween(1, 5)];
+        for (int i = 0; i < indicesPrivileges.length; i++) {
+            indicesPrivileges[i] = RoleDescriptor.IndicesPrivileges.builder()
+                .privileges(randomSubsetOf(randomIntBetween(1, 5), IndexPrivilege.names()))
+                .indices(randomList(1, 3, () -> randomAlphaOfLengthBetween(2, 8) + (randomBoolean() ? "*" : "")))
+                .build();
+        }
+        return indicesPrivileges;
+    }
+
+    private static RoleDescriptor.ApplicationResourcePrivileges[] randomApplicationResourcePrivileges() {
+        RoleDescriptor.ApplicationResourcePrivileges[] appPrivileges = new RoleDescriptor.ApplicationResourcePrivileges[randomIntBetween(
+            1,
+            5
+        )];
+        for (int i = 0; i < appPrivileges.length; i++) {
+            appPrivileges[i] = RoleDescriptor.ApplicationResourcePrivileges.builder()
+                .application(randomAlphaOfLength(1).toLowerCase(Locale.ROOT) + randomAlphaOfLengthBetween(2, 8))
+                .resources(randomList(1, 3, () -> randomAlphaOfLengthBetween(5, 7) + (randomBoolean() ? "*" : "")))
+                .privileges(generateRandomStringArray(3, 7, false, false))
+                .build();
+        }
+        return appPrivileges;
+    }
+
+    private PrivilegesToCheck randomInvalidPrivilegesToCheckRequest() {
+        return randomFrom(
+            new PrivilegesToCheck(randomBoolean() ? null : new String[0], randomIndicesPrivileges(), randomApplicationResourcePrivileges()),
+            new PrivilegesToCheck(
+                randomClusterPrivileges(),
+                randomBoolean() ? null : new RoleDescriptor.IndicesPrivileges[0],
+                randomApplicationResourcePrivileges()
+            ),
+            new PrivilegesToCheck(
+                randomClusterPrivileges(),
+                randomIndicesPrivileges(),
+                randomBoolean() ? null : new RoleDescriptor.ApplicationResourcePrivileges[0]
+            )
+        );
+    }
```

Comment:
If the randomisation picks the empty array, why is the overall `PrivilegesToCheck` invalid? That is, isn't 
`new PrivilegesToCheck(new String[0], randomIndicesPrivileges(), randomApplicationResourcePrivileges())` valid?

Reply:
It was a mistake, I pushed [f59f562](https://github.com/elastic/elasticsearch/pull/85898/commits/f59f562c31dd6f1bafa34089a97d0faf4af05719)